rest.go

```diff
@@ -89,9 +89,23 @@ func makeRESTAuth(authToken string) string {
 	return fmt.Sprintf("ModeCloud %s", authToken)
 }
 
-func makeRESTCall(method string, path string, authToken string, data interface{}, resData interface{}) error {
+func makeRESTCall(method string, path string, dc *DeviceContext, data interface{}, resData interface{}) error {
 	url := makeRESTCallURL(path)
 
+	authToken := ""
+	if dc != nil {
+		authToken = dc.AuthToken
+		if dc.TLSClientAuth {
+			config, err := dc.buildConfig()
```

Comment:
I prefer not having to call `dc.buildConfig()` every time we make a REST call. We can initialize this when the DeviceContext was first created.

Maybe we should add a function to create DeviceContext for TLS client and properly initialize it.

Reply:
I changed `dc.buildConfig()` to `dc.SetPKCS12ClientCertificate()` to be able to initialize device context when the DeviceContext was first created.